### DISCUSSION TITLE
Struggling to connect rTorrent to AutoBrr, what is the correctly formatted host url I need to use?

### DISCUSSION THREAD

**Discussion by Quantumatum** [crazy-max/docker-rtorrent-rutorrent#334] [Q&A]
Hi, i am trying to add rTorrent as a client to AutoBrr as instructed here: https://autobrr.com/configuration/download-clients/dedicated/#rtorrent--rutorrent

I can manually navigate to the to ruTorrent GUI via 10.0.0.52:8080 and have successfully setup basic auth.

However, none of the examples given in AutoBrr are working, they all fail when clicking 'test' and do not connect.

In the past I have been able to complete this step without issue when using rTorrent on a seedbox, so I'm not sure why it's not working in this case. I suspect it's something very trivial

Any help appreciated, thank you

```
docker run
  -d
  --name='rutorrent-crazymax'
  --net='br0'
  --ip='10.0.0.52'
  -e TZ="Europe/London"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="quantumbox"
  -e HOST_CONTAINERNAME="rutorrent-crazymax"
  -e 'UDP_PORT_6881'='6882'
  -e 'TCP_PORT_8080'='8095'
  -e 'RT_INC_PORT'='50000'
  -e 'TCP_PORT_50000'='50000'
  -e 'PGID'='100'
  -e 'PUID'='99'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:8080]/'
  -l net.unraid.docker.icon='https://github.com/superboki/ressources/blob/main/rutorrent-crazymax/rutorrent-crazymax.png?raw=true'
  -v '/mnt/user/data/torrents/downloads/rtorrent/':'/downloads':'rw'
  -v '/mnt/user/appdata/rutorrent-crazymax/data':'/data':'rw'
  -v '/mnt/user/appdata/rutorrent-crazymax/passwd':'/passwd':'rw'
  --ulimit nproc=65535
  --ulimit nofile=32000:40000 'crazymax/rtorrent-rutorrent'
02c711c7922026364b15f44c2e2ee121aa334ce26874a8bb076e7fec119bbfa2
```

**Comment by waz1500** [comment]
The unraid template you used missed the variable `XMLRPC_PORT` which will be needed for this to work. You'll need to expose and add the xmlrpc port to your configuration. 
`- XMLRPC_PORT=8000`

A good example of working compose can be found here. 
https://github.com/crazy-max/docker-rtorrent-rutorrent/tree/master/examples/compose

This container does work with services like sonarr/radarr utilising the SCGI socket but it has to be exposed first. 
It will also by default be unsecure. Be sure to add authentication once it's working as you did for the rutorrent interface and as described here. 
https://github.com/crazy-max/docker-rtorrent-rutorrent#xmlrpc-through-nginx

**Comment by Quantumatum** [comment]
> The unraid template you used missed the variable `XMLRPC_PORT` which will be needed for this to work. You'll need to expose and add the xmlrpc port to your configuration. `- XMLRPC_PORT=8000`
> 
> A good example of working compose can be found here. https://github.com/crazy-max/docker-rtorrent-rutorrent/tree/master/examples/compose
> 
> This container does work with services like sonarr/radarr utilising the SCGI socket but it has to be exposed first. It will also by default be unsecure. Be sure to add authentication once it's working as you did for the rutorrent interface and as described here. https://github.com/crazy-max/docker-rtorrent-rutorrent#xmlrpc-through-nginx

Thank you @waz1500, you where bang on - I just got it working by adding the `XMLRPC_PORT` variable with a value of 8000. All i had to do then was feed the following host url to AutoBrr and it immediately worked:

http://username:password@host_or_docker_ip:8000/rutorrent/plugins/httprpc/action.php
